doc/nrf/libraries/modem/location.rst

```diff
@@ -0,0 +1,198 @@
+
+Location library provides a convenient API for clients to request devices location.
+The client can determine the preference order of the positioning methods to be used along with some other configuration information.
+Once a method fails to provide location, Location library will perform fallback to the next preferred method.
+
+Supported positioning methods are GNSS, cellular and WiFi positioning.
+Both cellular and WiFi positioning take advantage of the base stations seen with corresponding technology, and then utilizing web services for retrieving the location.
+GNSS positioning uses satellites to compute the location of the device. Location library can utilize assistance data (A-GPS and P-GPS) to find the satellites faster.
+
+Implementation
+==============
+
+Location library has a compact API and then location core that handles location method independent part of the functionality such as fallback to the next preferred method and timeouts.
+Each location method has its own implementation for the location retrieval:
+
+* GNSS positioning
+   * :ref:`gnss_interface` for getting satellite information
+   * A-GPS/P-GPS are managed either with
+      * :ref:`lib_nrf_cloud_agps` and :ref:`lib_nrf_cloud_pgps`, or
+      * the application utilizes some other source for the data and uses :c:func:`location_agps_data_process` to pass the data to Location library
+   * A-GPS/P-GPS data format must be as received from :ref:`lib_nrf_cloud_agps`
+   * A-GPS/P-GPS data transport for :ref:`lib_nrf_cloud_agps` and :ref:`lib_nrf_cloud_pgps` can be configured to be either MQTT (:kconfig:`CONFIG_NRF_CLOUD_MQTT`) or REST (:kconfig:`CONFIG_NRF_CLOUD_REST`)
+* Cellular positioning
+   * :ref:`lte_lc_readme` for getting visible cellular base stations
+   * :ref:`lib_multicell_location` for sending cell information to the selected location service and getting calculated location back to the device
+      * Service is selected in :c:struct:`loc_method_config` when requesting location
+      * Data transport for the service is REST
+      * Available services are `nRF Cloud Location Services`_, `HERE Positioning`_, `Skyhook Precision Location`_ and `Polte Location API`_
+* WiFi positioning
+   * Zephyr's Network Management API :ref:`zephyr:net_mgmt_interface` for getting visible WiFi access points
+   * Sending access point information to the selected location service and getting calculated location back to the device
+      * Location library has implementation for WiFi location services
+      * Service is selected in :c:struct:`loc_method_config` when requesting location
+      * Data transport for the service is REST
+      * Available services are `nRF Cloud Location Services`_, `HERE Positioning`_ and `Skyhook Precision Location`_
+
+Supported features
+==================
+
+TODO: Not really sure what to put into this mandatory section given Implementation and Configuration sections has a lot of information and if something is missing, that can be added.
```

Comment:
@peknis You could comment this one too. I can repeat some stuff from Overview and Implementation sections but is there something different that I should write here?

Reply:
Removing Supported features section.